File: e-arbiter-tournament-results/src/main/java/pl/cyganki/results/controller/ResultsController.java
package pl.cyganki.results.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import pl.cyganki.results.model.database.Result;
import pl.cyganki.results.repository.ResultRepository;

import java.util.List;

@RestController
@RequestMapping("/api")
public class ResultsController {

    private ResultRepository resultRepository;

    @Autowired
    public ResultsController(ResultRepository resultRepository) {
        this.resultRepository = resultRepository;
    }

    @GetMapping("/all")
    public List<Result> getAllResults() {
        return resultRepository.findAll();
    }
}

File: e-arbiter-tournament-results/src/main/kotlin/pl/cyganki/results/controller/inner/ResultsController.kt
package pl.cyganki.results.controller.inner

import io.swagger.annotations.ApiOperation
import org.springframework.web.bind.annotation.*
import pl.cyganki.results.service.ResultService
import pl.cyganki.utils.model.tournamentresults.UsersTasksList

@RestController
@RequestMapping("/inner/results")
class ResultsController(private val resultService: ResultService) {

    // due to problems with passing bigger object in feign client, this request is made as a post
    @PostMapping("/{id}")
    @ApiOperation("Returns all results for tournament by passed id")
    fun getTournamentResults(@PathVariable("id") tournamentId: String, @RequestBody usersAndTasks: UsersTasksList)
            = resultService.getTournamentResults(tournamentId, usersAndTasks)

    // due to problems with passing bigger object in feign client, this request is made as a post
    @PostMapping("/user-place/{id}/{user-id}")
    @ApiOperation("Returns user's position with passed user-id in tournament with passed id")
    fun getUserPlaceInTournament(@PathVariable("id") tournamentId: String, @PathVariable("user-id") userId: Long, @RequestBody usersAndTasks: UsersTasksList)
            = resultService.getUserPlaceInTournament(tournamentId, userId, usersAndTasks)

}